pkg/messagebroker/pulsar.go

```diff
@@ -3,44 +3,128 @@ package messagebroker
 import (
 	"context"
 	"time"
+
+	"github.com/apache/pulsar-client-go/pulsar"
 )
 
 // PulsarBroker for pulsar
 type PulsarBroker struct {
-	Ctx    context.Context
-	Config *BrokerConfig
+	Ctx      context.Context
+	Config   *BrokerConfig
+	Consumer pulsar.Consumer
+	Producer pulsar.Producer
+}
+
+// NewPulsarConsumer returns a pulsar consumer
+func NewPulsarConsumer(ctx context.Context, bConfig *BrokerConfig) (Consumer, error) {
+	client, err := pulsar.NewClient(pulsar.ClientOptions{
+		URL:               bConfig.Brokers[0],
```

Comment:
put a validation saying pulsar broker list should have only 1 member during init

Reply:
check in registry